### ISSUE TITLE
Error: Minified React error #130; visit https://reactjs.org/docs/error-decoder.html?invariant=130...

### ISSUE THREAD

**Issue by JR-Carroll** [netlify/netlify-cms#4692]
**Describe the bug**
Having blended the netlify tutorial and sourcing the netlify starter blog sample project into my gatsby website, I have all of it working except the 'preview' portion of the WYSIWYG.  

What I'm really looking for is a way to do greater debug on this, as I doubt I'll be able to create a repo for you all to reproduce this.  Right now, all I see is "There's been an error - please open an issue on GitHub." and there isn't any other real substantial details for me to continue to review.

Based on my own research this is likely a problem with page template and/or the wrapper for the preview, but I can't source the problem since everything is minified.  (Following the various web articles to --no-uglify and.or to modify gatsby-node.js webPackConfig block with optimization minifiy: false didn't do anything -- I suspect 'me' error and limited understanding of webpack + gatsby).  

**To Reproduce**

**Expected behavior**

**Screenshots**
![image](https://user-images.githubusercontent.com/2183723/101323967-b4503480-3826-11eb-9356-f99b35559f4e.png)

**Applicable Versions:**
 - Netlify CMS version: `netlify-cms-app@2.14.3`
 - Git provider: `git-gateway`
 - Browser version: `Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/87.0.4280.88 Safari/537.36`

**CMS configuration**
```
publish_mode: simple
slug:
  encoding: unicode
  clean_accents: false
  sanitize_replacement: "-"
backend:
  name: git-gateway
  branch: master
  commit_messages: null
  create: Create {{collection}} “{{slug}}”
  update: Update {{collection}} “{{slug}}”
  delete: Delete {{collection}} “{{slug}}”
  uploadMedia: "[skip ci] Upload “{{path}}”"
  deleteMedia: "[skip ci] Delete “{{path}}”"
media_folder: static/img
public_folder: /img
collections:
  - sortable_fields:
      - commit_date
      - title
      - date
      - commit_author
      - description
    view_filters: []
    meta: {}
    name: blog
    slug: "{{year}}-{{month}}-{{day}}-{{slug}}"
    folder: src/pages/content/blog
    view_groups: []
    publish: true
    label: Blog
    fields:
      - label: Template Key
        name: templateKey
        widget: hidden
        default: blog-post
      - label: Title
        name: title
        widget: string
      - label: Publish Date
        name: date
        widget: datetime
      - label: Description
        name: description
        widget: text
      - label: Featured Post
        name: featuredpost
        widget: boolean
      - label: Featured Image
        name: featuredimage
        widget: image
      - label: Body
        name: body
        widget: markdown
      - label: Tags
        name: tags
        widget: list
    create: true

```

**Additional context**
Any thoughts on how I can get greater visibility into what the problem is under the hood?  aka a '-v' on the build =P or some other way to source what line(s) are causing this error?

**Comment by erezrokah** [comment]
Hi @JR-Carroll, following the link https://reactjs.org/docs/error-decoder.html/?invariant=130&args[]=undefined&args=[] shows some more information on the error.

We also have a debugging guide here:
https://github.com/netlify/netlify-cms/blob/master/CONTRIBUTING.md#debugging

> Based on my own research this is likely a problem with page template and/or the wrapper for the preview

This is correct. My guess is that you're trying to render something that React can't render (possibly `undefined`).
I recommend adding some logs in the custom preview component to see if the data you're getting matches expectations.

**Comment by farukrana** [comment]
issue

**Comment by jcarroll-ceo** [comment]
Thank you, @erezrokah // the problem was sourced to the previewTemplate I imported.  

To others who find this:

1. If you did your own research, you probably saw others having similar, but not identical issues, and they all have different (but often similar) causes.
2. Notes of takeaway:  As reported in various other pages, you CANNOT do typical 'react stuff' in the previewTemplates.  In my particular case, I added `{console.log()}` statements to the top of the previewTemplate to see what props/unpacked props were being passed into the template.  Turns out it was a ton of 'undefined's. Whelp, that'd break it for sure!
3.  To fix this I backed up in the stack a bit to find out who was responsible for passing those props and it turns out higher in the call chain it was related to my Layout component, which was doing a `graphql` call to get the sites main logo.  <--- this is the normal 'react stuff' I mentioned before.  Taking this out (as well as some other components that were trying to do things that the previewTemplate doens't support) fixed my issue.
4.  Now I am left with having a `*.js` minimal file to drive the 'preview', and then I'll have a separate file to drive the build/compile step for production, in this way I can have everything I've ever wanted out of life =P.

TL;DR; it IS possible to copy the configurations out of the sample netlify cms gatsby project (obviously making modifications to fit your project along the way), but I HIGHLY recommend that you go through the Netlify CMS tutorial instead, AND THEN, use the starter gatsby project to reference how you want to evolve your own site.  

The error above, as were most of integration woes, were mostly caused by the `config.yml` file, the file structure (aka src>img, and public>images and src>cms>images and just understanding how to lay it all out), and THIS issue (which was entirely non-obvious) really was related to unsupported React "stuff" in the previewTemplate -- deleting everything in the return/render and starting from scratch with console.log's helped to find this issue!

You WILL get undefined back (as a String) if something higher in the call chain doesn't work correctly, like inline StaticQueries and graphql statements.  

To the NetlifyCMS-team:  Love the CMS, thank you!  Im looking forward to using it.  If it's possible, as a feature request, there are SO many ways to screw up this integration that if we had a way to `yarn install netlify-cms` and then to use netlify-cli to run a config server to walk through some of this, it would have saved me 3 days of debugging and hacking my code to wrench this into my project (again, not your problem, that's on me, but I've seen enough blogs + bugs + requests for ease of integration over the last 3 days that I think this would help people get it fit into their project).  :two-cents: 🎤 -drop.

Thanks again
